src/main/java/edu/umich/its/spe/SPEEsbImpl.java

```diff
@@ -112,19 +146,30 @@ public WAPIResultWrapper putGradeViaESB(HashMap<String, String> value) {
 		.append("/Score/")
 		.append(value.get("SCORE"));
 
-		M_log.debug("putGrades: value:["+value.toString()+"]");
-		M_log.debug("putGrades: request url: ["+url.toString()+"]");
-		M_log.debug("putGrades: headers: ["+headers.toString()+"]");
+		M_log.debug("putGrades: value:[" + value.toString() + "]");
+		M_log.debug("putGrades: request url: [" + url.toString() + "]");
+		M_log.debug("putGrades: headers: [" + headers.toString() + "]");
 
 		WAPI wapi = new WAPI(value);
+		WAPIResultWrapper wrappedResult = null;
 
-		WAPIResultWrapper wrappedResult = wapi.doPutRequest(url.toString(),headers);
+		// For testing allow skipping update of MPathways
+		/// check for not true.
+		M_log.debug("skipiGradeUpdate: {}",value.get("skipGradeUpdate"));
+		if (value.get("skipGradeUpdate") == null || !"true".equals(value.get("skipGradeUpdate").toLowerCase())) {
+			wrappedResult = wapi.doPutRequest(url.toString(), headers);
+		} else {
+			String msg = "{error: " + SKIP_GRADE_UPDATE + " for " + value.get("UNIQNAME") + "}";
+			M_log.error("skip msg: " + msg);
+			wrappedResult = new WAPIResultWrapper(WAPI.HTTP_UNKNOWN_ERROR, SKIP_GRADE_UPDATE, new JSONObject(msg));
```

Comment:
WAPI shouldn't have application specific / special error codes.  It should be general enough to make it easy to use the ESB from Java.  It can't know anything about the application that calls it.